### PR TITLE
rerender conda-smity 3.1.1

### PR DESCRIPTION
**Pull request by looooo**
Fixes https://github.com/conda-forge/vtk-feedstock/issues/47

- using https://github.com/conda-forge/vtk-feedstock/pull/50
- rerendering with conda-smithy 3.1.1

### CODE REVIEW

**Comment by jakirkham** [comment]
Thanks for doing this, @looooo. Also thanks @isuruf for doing all the work necessary to get it to this point (which was not insignificant for those unaware).

Looks like one of the CircleCI macOS builds passed. Let's see how the rest do.

File: conda-forge.yml [hunk 1]
```diff
@@ -1,9 +1,8 @@
...
-- [[VTK_WITH_OSMESA, False]]
-- [[VTK_WITH_OSMESA, True]]
 travis:
   secure:
     BINSTAR_TOKEN: fhS9RaksW9C7p2Rw+f47sjuEV4IWQBG8C5YYCfxAIMTBEpdLcAeDYJshjchdn7qpQf8J3vOidWGnDJ5EetWvW9sHfcahL3KKLYEqTqKV066lHQkGl5TfT+ehYimrOKP0XhS2Y472FF3WuTARkZKSGFduKNfbgwLIus0udST4oHPlO/mdnvRO73W0GeU+QxH/y7RTWhjUIFI7e+9LWlv3B3pyaoiwyIr8IXrgPlv3szq8ip36knaLs9EG47+NgZn2mltXgIn0fOW/b4ljaispw7hSCGBvSEuitU23d0aTwI5rFRDdIpWG7hkStLrBZTBZEH76Z7ZlLoza1TYElaZJO4tJVhpKQ5kF8ECXSr6usi3p4i3nmOTOr31uXiZKfpptiYGXKg4Ckc60QKyl8FXlVoCCWMn20KsNepwDhavf+nptqf7tpo+gb2BRZp5QsHxnNA7rOzeYb49YENdgMRSoI7YTXqf5uEB7I7cZTQijnyKHE4O6IyQlLVT5GoIBfNp7Kr6UuAK+hm//2scUXBXHOlapJMteD/pXJ1ma31VOnBaqX14f5fXjrnfxdfgVuLOcBXUfc6hmHPCjTghQL3MudgzXuztIAmB4RJAABg1N0NlnE1qGRh+xLwzsziN3MPUDO5ZwGYQv/WpLSMqar7aegl31D4YLkmQAml1Tx0WTlrM=
 appveyor:
   secure:
     BINSTAR_TOKEN: ipv/06DzgA7pzz2CIAtbPxZSsphDtF+JFyoWRnXkn3O8j7oRe3rzqj3LOoq2DZp4
+
+osx_provider: circle
```
[the remark this answers is not in the document]
Comment (jakirkham) [reply conda-forge.yml]: Went ahead and this for simplicity. Sadly (but unsurprisingly) CircleCI ignored the skip directive.

**Comment by isuruf** [comment]
I cancelled the CI to avoid wasting 9 hours of CI time

**Comment by jakirkham** [comment]
Thank you. Sorry for leaving those up there.

**Comment by looooo** [comment]
so this works. great. thanks for your efforts @isuruf to make this possible.
regarding the wrong syntax in conda-forge.yml:  I applied the correct syntax before rerendering. But the change was not included in the commit. no idea.

**Comment by looooo** [comment]
@jakirkham how to move forward?
I would love to see this merged, as it is one of the main show-stoppers I currently have.

**Comment by jakirkham** [comment]
Thanks @looooo. Have not forgotten. ;)

So have added a fixes note for the macOS build issue. Will ping some people over there to review. Expect this will be merged in short order. :)

**Comment by jakirkham** [comment]
Should add this does appear to be using the right macOS SDK AFAICT. Picks up OpenGL from it for instance.

**Comment by jakirkham** [comment]
Planning on merging tomorrow evening if there are no comments.

**Comment by jakirkham** [comment]
Hmm...seeing a new error cropping up from `conda-build`. Seems to be treating `jpeg` in a weird way. Is this fixed with `conda-build` version `3.9.2`, @msarahan? If so, would it be alright to merge PR ( https://github.com/conda-forge/conda-build-feedstock/pull/45 )?

```python
Traceback (most recent call last):
  File "/opt/conda/lib/python3.6/site-packages/conda_build/environ.py", line 706, in get_install_actions
    actions = install_actions(prefix, index, specs, force=True)
  File "/opt/conda/lib/python3.6/site-packages/conda/common/io.py", line 46, in decorated
    return f(*args, **kwds)
  File "/opt/conda/lib/python3.6/site-packages/conda/plan.py", line 541, in install_actions
    txn = solver.solve_for_transaction(prune=prune, ignore_pinned=not pinned)
  File "/opt/conda/lib/python3.6/site-packages/conda/core/solve.py", line 505, in solve_for_transaction
    force_remove, force_reinstall)
  File "/opt/conda/lib/python3.6/site-packages/conda/core/solve.py", line 438, in solve_for_diff
    final_precs = self.solve_final_state(deps_modifier, prune, ignore_pinned, force_remove)
  File "/opt/conda/lib/python3.6/site-packages/conda/core/solve.py", line 318, in solve_final_state
    conflicting_specs = r.get_conflicting_specs(tuple(final_environment_specs))
  File "/opt/conda/lib/python3.6/site-packages/conda/resolve.py", line 705, in get_conflicting_specs
    reduced_index = self.get_reduced_index(specs)
  File "/opt/conda/lib/python3.6/site-packages/conda/resolve.py", line 254, in get_reduced_index
    specs, features = self.verify_specs(specs)
  File "/opt/conda/lib/python3.6/site-packages/conda/resolve.py", line 172, in verify_specs
    raise ResolvePackageNotFound(bad_deps)
conda.exceptions.ResolvePackageNotFound: 
  - jpeg[version='>=9,<10.0a0']
  - vtk==8.1.0=py27he461554_202
  - jpeg[version='>=9,<10.0a0']

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/conda/bin/conda-build", line 6, in <module>
    sys.exit(conda_build.cli.main_build.main())
  File "/opt/conda/lib/python3.6/site-packages/conda_build/cli/main_build.py", line 420, in main
    execute(sys.argv[1:])
  File "/opt/conda/lib/python3.6/site-packages/conda_build/cli/main_build.py", line 411, in execute
    verify=args.verify)
  File "/opt/conda/lib/python3.6/site-packages/conda_build/api.py", line 199, in build
    notest=notest, need_source_download=need_source_download, variants=variants)
  File "/opt/conda/lib/python3.6/site-packages/conda_build/build.py", line 2123, in build_tree
    test(pkg, config=metadata.config, stats=stats)
  File "/opt/conda/lib/python3.6/site-packages/conda_build/build.py", line 1878, in test
    channel_urls=tuple(metadata.config.channel_urls))
  File "/opt/conda/lib/python3.6/site-packages/conda_build/environ.py", line 708, in get_install_actions
    raise DependencyNeedsBuildingError(exc, subdir=subdir)
conda_build.exceptions.DependencyNeedsBuildingError: Unsatisfiable dependencies for platform linux-64: {'vtk==8.1.0=py27he461554_202', "jpeg[version='>=9,<10.0a0']"}
```

ref: https://circleci.com/api/v1.1/project/github/conda-forge/vtk-feedstock/395/output/104/0?file=true

cc @stuarteberg (as we discussed something similar the other day)

**Comment by stuarteberg** [comment]
The problem is that `- jpeg 9*` is interpreted as `- jpeg >=9,<10.0a0`, which *looks* correct, but `jpeg` uses a weird versioning scheme.  The current version of `jpeg` is `9b`, which is interpreted as `9.beta` (more-or-less) by PEP440.  According to PEP440, `9b` < `9.0` and therefore does not satisfy `>=9,<10.0a0`.

The quick fix is to change the `jpeg` requirement in this recipe: `>=9a,<10` should work.

**Comment by jakirkham** [comment]
We might be able to drop the pinning altogether since PR ( https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/46 ) went in. Though it would be good if someone can confirm whether this is true or not. Will work up a PR for discussion.

ref: https://conda.io/docs/user-guide/tasks/build-packages/variants.html#pinning-at-the-variant-level

**Comment by jakirkham** [comment]
Restarting the builds on `master` to see if recent `conda-build` releases fixed the previously seen issue.

cc @ivoflipse

**Comment by jakirkham** [comment]
Looks like they all built.

**Comment by looooo** [comment]
great! thanks for your work @jakirkham @isuruf

**Pull request merged**